src/test/java/com/treasuredata/client/TestTDClient.java

```diff
@@ -1958,6 +1959,50 @@ public void testImportBytesWithId()
         assertEquals(result.getUniqueId(), "4288048cf8f811e88b560a87157ac806");
     }
 
+    @Test
+    public void testGetFederatedQueryConfigsWhenEmpty()
```

Comment:
Could you add test cases that the endpoint return 4xx status code?
The endpoint returns `401 and 403`.

Reply:
It is not very convenient to simulate 401 and 403. I added a test case of 401, while it depends on the internal behaviour. 
To support 401 and 403 in the test case, we need to configure the account of staging-aws used by the unit tests in a way to trigger 401 and 403 (Some junit tests in this class actually sends real request to staging-aws api3), which I feel might be a little too much.

Reply:
sure, I see.